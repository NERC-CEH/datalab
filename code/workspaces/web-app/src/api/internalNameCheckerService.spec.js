import mockClient from './graphqlClient';
import internalNameCheckerSevice from './internalNameCheckerService';

jest.mock('./graphqlClient');

describe('internalNameCheckerService', () => {
  beforeEach(() => mockClient.clearResult());

  describe('checkNameUniqueness', () => {
    it('should build the correct query and unpack the results', () => {
      const queryParams = { projectKey: 'testproj', name: 'validateName' };
      mockClient.prepareSuccess({ checkNameUniqueness: 'expectedValue' });

      return internalNameCheckerSevice.checkNameUniqueness(queryParams.projectKey, queryParams.name).then((response) => {
        expect(response).toEqual('expectedValue');
        expect(mockClient.lastQuery()).toMatchSnapshot();
        expect(mockClient.lastOptions()).toEqual(queryParams);
      });
    });

    it('should throw an error if the query fails', () => {
      mockClient.prepareFailure('error');

      return internalNameCheckerSevice.checkNameUniqueness('validateName').catch((error) => {
        expect(error).toEqual({ error: 'error' });
      });
    });
  });
});
