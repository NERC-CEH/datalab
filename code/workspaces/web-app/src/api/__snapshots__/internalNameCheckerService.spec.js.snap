// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`internalNameCheckerService checkNameUniqueness should build the correct query and unpack the results 1`] = `
"
    CheckInternalName($projectKey: String!, $name: String!) {
      checkNameUniqueness(projectKey: $projectKey, name: $name)
    }"
`;
